### ISSUE TITLE
[ENH] Extend forecasting benchmark to global setting

### ISSUE THREAD

**Issue by benHeid** [sktime/sktime#6661]
As far as I know, currently the forecasting benchmarking (as well as evaluate) is not working for global setting. I.e., fitting a forecasting model on time series 1:n and evaluating it on n+1:m with n<m. 

@fkiraly please mention if it is already implemented.

**Comment by fkiraly** [comment]
No, not implemented. For this, `evaluate` would have to be adapted I think.

I would be surprised if it worked out of the box - it *could* be in theory because there is an `InstanceSplitter` which splits a collection of time series by instance, so it might be if we pass that it works. Even if, this is perhaps not the best user experience.

**Comment by benHeid** [comment]
@Xinyu-Wu-0000 I think you are working on this. Thus, could you comment here, so that I can assign you?

**Comment by Xinyu-Wu-0000** [comment]
> @Xinyu-Wu-0000 I think you are working on this. Thus, could you comment here, so that I can assign you?

Yeah, you could assign me on the PR #6774 too.
